src/features/filter.ts

```diff
@@ -1,10 +1,49 @@
-export const actions = { /* put action creators here */};
+import { Status } from '../types/Status';
 
-const filterReducer = () => {
-  return {
-    query: '',
-    status: 'all',
-  };
+type AllActions = { type: 'ALL', payload: string };
+type ActiveActions = { type: 'ACTIVE', payload: string };
+type CompletedActions = { type: 'COMPLETED', payload: string };
+type InputActions = { type: 'INPUT', payload: string };
+
+type Action = AllActions | ActiveActions | CompletedActions | InputActions;
+
+const all = (query: string): Action => ({ type: 'ALL', payload: query });
+const active = (query: string): Action => ({ type: 'ACTIVE', payload: query });
+const completed = (query: string): Action => (
```

Comment:
better to create one action instead of these three and pass in this action in payload status value